Parsers/ASimNetworkSession/Parsers/vimNetworkSessionWatchGuardFirewareOS.yaml

```diff
@@ -0,0 +1,182 @@
+  - Title: WatchGuard Help Center
+    Link: https://www.watchguard.com/help/docs/help-center/en-US/Content/en-US/Fireware/_intro/fireware_help_front.html?tocpath=Fireware%7C_____0
+Description: |
+  This ASIM parser supports filtering and normalizing WatchGuard Fireware OS logs produced by the Microsoft Sentinel Syslog connector to the ASIM Network Session normalized schema.
+ParserName: vimNetworkSessionWatchGuardFirewareOS                                  
+EquivalentBuiltInParser: _Im_NetworkSession_WatchGuardFirewareOS                                  
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventLookup=datatable(DvcAction:string,EventResult:string,EventSeverity:string)
+  [
+      "Allow","Success","Informational"
+      , "Deny","Failure","Low"
+  ];
+  let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let dst_or_any=set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+  let Parser=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), srcipaddr_has_any_prefix:dynamic=dynamic([]), dstipaddr_has_any_prefix:dynamic=dynamic([]), ipaddr_has_any_prefix:dynamic=dynamic([]), dstportnumber:int=int(null), hostname_has_any:dynamic=dynamic([]), dvcaction:dynamic=dynamic([]), eventresult:string='*', disabled:bool=false){
+  let AllLogs = Syslog
+  | where not(disabled)
+  | where (isnull(starttime) or TimeGenerated>=starttime) and (isnull(endtime) or TimeGenerated<=endtime)
+  | where SyslogMessage has_any('msg_id="3000-0148"' 
+      , 'msg_id="3000-0149"' 
+      , 'msg_id="3000-0150"'
+      , 'msg_id="3000-0151"'
+      , 'msg_id="3000-0173"'
+  )
+  | project TimeGenerated, SyslogMessage, HostName;
+  let WithFQDN = AllLogs
+  | where SyslogMessage has "fqdn_dst_match=" and SyslogMessage !has "3000-0151" and SyslogMessage !has "igmp"
+  | parse kind=regex flags=U SyslogMessage with * @'msg_id="' MessageID @'"\sfqdn_dst_match="' DstDomain @'"\s' DvcAction @"\s" RuleName @"\s\d{2,5}\s" NetworkProtocol @"\s\d+\s\d+\s" SourceIP @"\s" DestinationIP @"\s" SrcPortNumber:int @"\s" DestinationPort:int @"\s" *
+  | where  (isnull(dstportnumber) or DestinationPort==dstportnumber)
+  | extend temp_SrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any)
+  , temp_DstMatch=has_any_ipv4_prefix(DestinationIP,dst_or_any)

```

Comment:
Out of curiosity, why did you use DestinaitonIP throughout and then project-rename at the end?